src/components/Checkout.jsx

```diff
@@ -0,0 +1,80 @@
+import { useState } from "react";
+
+export default function Checkout({ setCartArr }) {
+  const inputs = {
+    firstName: "",
+    lastName: "",
+    email: "",
+    zipCode: "",
+  };
+  const [inputState, setInputState] = useState(inputs);
+  function renderTextChange(e, inputKey) {
+    const inputClone = { ...inputState };
```

Comment:
could also be one line `const inputClone = { ...inputState, [inputKey]: e.target.value }`